src/CodeGen.cpp

```diff
@@ -174,7 +174,7 @@ bool CodeGen::llvm_NVPTX_enabled = false;
 bool CodeGen::llvm_Mips_enabled = false;
 
 void CodeGen::compile(Stmt stmt, string name,
-                      const vector<Argument> &args,
+                      const ArgInfo &arg_info,
```

Comment:
Why use a struct instead of just two arguments: 'vector<Argument> args, int num_outputs'?

It would make it easier to call (no need to set up a struct just to call this function), and a more selfish motivation is that this PR is going to be two more very large merges with codegen_compile_refactor and modules. This would at least minimize the changes somewhat.